ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -0,0 +1,822 @@
+              Seq.Empty -> -- No updates to apply. So the current ledger state is
+                          -- valid from the end of the last snapshot to the first
+                          -- scheduled update.
+                slotBounded lb ub dsNow
+              toApply@(_ Seq.:|> la) -> slotBounded lb (convertSlot . V.Scheduling.sdSlot $ la)
+                $ foldl (\acc x -> Bimap.insert (V.Scheduling.sdDelegator x) (V.Scheduling.sdDelegate x) acc) dsNow toApply
+          else Nothing
+          where
+            lb = case ss of
+              _ Seq.:|> s -> max lvLB (sbUpper s)
+              Seq.Empty   -> lvLB
+            ub = case dsScheduled of
+              s Seq.:<| _ -> min lvUB (convertSlot $ V.Scheduling.sdSlot s)
+              Seq.Empty   -> lvUB
+    where
+      SecurityParam paramK = pbftSecurityParam . pbftParams . encNodeConfigP $ cfg
+      lvUB = SlotNo $ unSlotNo currentSlot + (2 * paramK)
+      lvLB = SlotNo $ if 2 * paramK > unSlotNo currentSlot then 0 else unSlotNo currentSlot - (2 * paramK)
+      dsNow = Delegation.unMap
+              . V.Interface.delegationMap
+              . CC.Block.cvsDelegationState
+              $ ls
+      dsScheduled = V.Scheduling.scheduledDelegations
+              . V.Interface.schedulingState
+              . CC.Block.cvsDelegationState
+              $ ls
+      currentSlot = convertSlot $ CC.Block.cvsLastSlot ls
+      containsSlot s sb = sbLower sb <= s && sbUpper sb >= s
+
+{-------------------------------------------------------------------------------
+  Mempool integration
+
+class UpdateLedger b => ApplyTx b where
+  -- | Generalized transaction
+  --
+  -- The mempool (and, accordingly, blocks) consist of "generalized
+  -- transactions"; this could be "proper" transactions (transferring funds) but
+  -- also other kinds of things such as update proposals, delegations, etc.
+  type family GenTx b :: *
+
+  -- | Apply transaction we have not previously seen before
+  applyTx :: GenTx b
+          -> LedgerState b
+          -> Except (LedgerError b) (LedgerState b)
+
+  -- | Re-apply a transaction
+  --
+  -- When we re-apply a transaction to a potentially different ledger state
+  -- expensive checks such as cryptographic hashes can be skipped, but other
+  -- checks (such as checking for double spending) must still be done.
+  reapplyTx :: GenTx b
+            -> LedgerState b
+            -> Except (LedgerError b) (LedgerState b)
+
+  -- | Re-apply a transaction to the very same state it was applied in before
+  --
+  -- In this case no error can occur.
+  --
+  -- See also 'ldbConfReapply' for comments on implementing this function.
+  reapplyTxSameState :: GenTx b -> LedgerState b -> LedgerState b
+-------------------------------------------------------------------------------}
+
+-- | Generalized transactions in Byron
+--
+-- TODO: This is still missing the other cases (this shouldn't be a newtype)
```

Comment:
Let's create a ticket for this, and refer to https://github.com/input-output-hk/cardano-ledger/issues/514 .